File: Safari-609.1.11.1/LayoutTests/http/tests/security/mixedContent/resources/subresource/protected-page.php
<?php
header("Cache-Control: no-store");
header("Connection: close");
if (!isset($_SERVER["PHP_AUTH_USER"])) {
    header("WWW-authenticate: Basic realm=\"" . $_SERVER["REQUEST_URI"] . "\"");
    header("HTTP/1.0 401 Unauthorized");
}
header("Content-Type: text/html");
?>
<!DOCTYPE html>
<html>
<body>
<p><?php if (isset($_SERVER["PHP_AUTH_USER"])) { ?>
Authenticated with username <?php echo htmlspecialchars($_SERVER["PHP_AUTH_USER"]); ?>.
<?php } else { ?>
Unauthorized.
<?php } ?></p>
<script>
if (window.testRunner)
    testRunner.notifyDone();
</script>
</body>
</html>

File: LayoutTests/http/tests/security/mixedContent/resources/subresource/protected-page.py
#!/usr/bin/env python3

import base64
import os
import sys

username = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')[0]
uri = os.environ.get('REQUEST_URI', '')

sys.stdout.write(
    'Cache-Control: no-store\r\n'
    'Connection: close\r\n'
    'Content-Type: text/html\r\n'
)

if not username:
    sys.stdout.write(
        'WWW-authenticate: Basic realm="{}"\r\n'
        'status: 401\r\n'.format(uri)
    )

sys.stdout.write(
    '\r\n<!DOCTYPE html>\n'
    '<html>\n'
    '<body>\n'
    '<p>'
)

if username:
    sys.stdout.write('Authenticated with username {}.'.format(username))
else:
    sys.stdout.write('Unauthorized.')

sys.stdout.write(
    '</p>\n'
    '<script>\n'
    'if (window.testRunner)\n'
    '    testRunner.notifyDone();\n'
    '</script>\n'
    '</body>\n'
    '</html>\n'
)